src/unordered_map.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2011 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NINJA_UNORDERED_MAP_H_
+#define NINJA_UNORDERED_MAP_H_
+
+#include "string_piece.h"
+
+#if __APPLE__ || (__cplusplus >= 201103L) || (_MSC_VER >= 1900)
```

Comment:
Should only be `__cplusplus >= 201103L`.

Reply:
No, that's not sufficient.

AppleClang doesn't ship with tr1 headers, regardless of the standard.

MSVC appears to have the same problem.

Reply:
This if branch is for the non-tr1 version though?

Reply:
That's correct.

If we only check for C++11 to determine whether to include <unordered_map>, then AppleClang and MSVC (even when compiling for C++98) will fail to find <tr1/unordered_map> in the #else below, because they do not ship that header.

Reply:
Shouldn't both default to >= C++11 anyway though?

Reply:
I don't know.

If I take away either of __APPLE__ or (_MSC_VER >= 1900), it won't build.

If I leave them in it does.

I'm not really interested in putting a lot of time into this, because I consider supporting anything older than C++11 a waste of time.

I'm submitting this PR to Ninja because I use this code (the non tr1 version) in my fork. I benefit by having a smaller set of patches to update against Ninja master.

If you have specific insights into how to improve this, I'm all ears, but I don't have either a Mac or a Windows computer, so it's already an uphill battle to figure out why it does or doesn't build when I change the flags.

My recommendation is to accept the change, and let other contributors point out problems if there are any. Ideally, with continuous integration tests that reproduce the failure automatically.

Reply:
I'll have a look when I have the time.

Reply:
> If I take away either of __APPLE__ or (_MSC_VER >= 1900), it won't build.

Regarding MSVC: Does it work with the CMake build? Because it should after 71c2cf3905a66f68409edde4745cb69ed967aefb.

Reply:
I used the GitHub Actions continuous integration to determine if things built.